src/transformers/integrations.py

```diff
@@ -1162,10 +1182,12 @@ def on_train_end(self, args, state, control, **kwargs):
     def __del__(self):
         self._stop_run_if_exists()
 
+        if self._volatile_checkpoints_dir is not None:
+            shutil.rmtree(self._volatile_checkpoints_dir, ignore_errors=True)
+
     def on_save(self, args, state, control, **kwargs):
-        if self._should_upload_checkpoint:
-            checkpoint_path = os.path.join(args.output_dir, f"checkpoint-{state.global_step}")
-            self._log_model_checkpoint(checkpoint_path)
+        if self._should_upload_checkpoint is not None:
```

Comment:
`_should_upload_checkpoint` is `bool`, should not be compared to `None`.

Reply:
Fixed